File: src/Python/modules/infrastructure/local_weather_provider.py
from modules.domain.user_cases.weather.get.ports.iweather_provider import (
    IWeatherProvider,
)
from modules.domain.user_cases.weather.weater_model import WeatherModel


class LocalWeatherProvider(IWeatherProvider):
    async def get_weater(self) -> "WeatherModel":
        return WeatherModel(is_raining=True)

File: src/C#/Pillont.Shodo.Infrastructure/Weather/LocalWeatherProvider.cs
using Pillont.Shodo.Domain.UseCases.Weather.Get.Ports;
using Pillont.Shodo.Domain.UseCases.Weather.Models;

namespace Pillont.Shodo.Infrastructure.Weather;

public class LocalWeatherProvider : IWeatherProvider
{
    private WeatherModel CurrentWeather { get; set; } = new WeatherModel() 
    { 
        IsRaining = true 
    };

    public Task<WeatherModel> GetAsync()
    {
        return Task.FromResult(CurrentWeather );
    }
}
